NetKet/Sampler/custom_sampler.hpp

```diff
@@ -0,0 +1,277 @@
+      throw InvalidInputError("MoveOperators is not an array");
+    }
+    CheckFieldExists(pars["Sampler"], "ActingOn");
+
+    if (!pars["Sampler"]["ActingOn"].is_array()) {
+      throw InvalidInputError("ActingOn is not an array");
+    }
+
+    auto jop = pars["Sampler"]["MoveOperators"].get<std::vector<MatType>>();
+    auto sites =
+        pars["Sampler"]["ActingOn"].get<std::vector<std::vector<int>>>();
+
+    if (sites.size() != jop.size()) {
+      throw InvalidInputError(
+          "The custom sampler definition is inconsistent (between "
+          "MoveOperators and ActingOn sizes); Check that ActingOn is defined");
+    }
+
+    std::set<int> touched_sites;
+    for (std::size_t c = 0; c < jop.size(); c++) {
+      // check if matrix of this operator is stochastic
+      bool is_stochastic = true;
+      bool is_complex = false;
+      bool is_definite_positive = true;
+      bool is_symmetric = true;
+      const double epsilon = 1.0e-6;
+      for (std::size_t i = 0; i < jop[c].size(); i++) {
+        double sum_column = 0.;
+        for (std::size_t j = 0; j < jop[c].size(); j++) {
+          if (std::abs(jop[c][i][j].imag()) > epsilon) {
+            is_complex = true;
+            is_stochastic = false;
+            break;
+          }
+          if (jop[c][i][j].real() < 0) {
+            is_definite_positive = false;
+            is_stochastic = false;
+            break;
+          }
+          if (std::abs(jop[c][i][j].real() - jop[c][j][i].real()) > epsilon) {
+            is_symmetric = false;
+            is_stochastic = false;
+            break;
+          }
+          sum_column += jop[c][i][j].real();
+        }
+        if (std::abs(sum_column - 1.) > epsilon) {
+          is_stochastic = false;
+        }
+      }
+      if (is_complex == true) {
+        InfoMessage() << "Warning: MoveOperators " << c
+                      << " has complex matrix elements" << std::endl;
+      }
+      if (is_definite_positive == false) {
+        InfoMessage() << "Warning: MoveOperators " << c
+                      << " has negative matrix elements" << std::endl;
+      }
+      if (is_symmetric == false) {
+        InfoMessage() << "Warning: MoveOperators " << c << " is not symmetric"
+                      << std::endl;
+      }
+      if (is_stochastic == false) {
+        InfoMessage() << "Warning: MoveOperators " << c << " is not stochastic"
+                      << std::endl;
+        InfoMessage() << "MoveOperators " << c << " is discarded" << std::endl;
+      }
+
+      else {
+        move_operators_.push_back(LocalOperator(hilbert_, jop[c], sites[c]));
+        for (std::size_t i = 0; i < sites[c].size(); i++) {
+          touched_sites.insert(sites[c][i]);
+        }
+      }
+    }
+
+    if (move_operators_.size() == 0) {
+      throw InvalidInputError("No valid MoveOperators provided");
+    }
+
+    if (static_cast<int>(touched_sites.size()) != hilbert_.Size()) {
```

Comment:
As a slight improvement, rather than checking the bare size of the support, here we should  check that touched_sites contains at least once all the sites of the lattice. I can do this change later